.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,15 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/universal:2",
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "stylelint.vscode-stylelint",
+        "esbenp.prettier-vscode",
+        "dbaeumer.vscode-eslint",
+        "streetsidesoftware.code-spell-checker",
+        "astro-build.astro-vscode"
+      ]
+    }
+  },
+  "postCreateCommand": "yarn install && yarn build"
```

Comment:
We don't need `yarn build` on creation, and it takes too long. `yarn install` should be enough.

```suggestion
  "postCreateCommand": "yarn install"
```

Reply:
Actually I was testing it out and found that `postCreateCommand` runs too late, after everything in the Codespace is initialized. [`onCreateCommand`](https://containers.dev/implementors/features/#lifecycle-hooks) should work better.

```
"onCreateCommand": "yarn install"
```

Reply:
made that change

Reply:
Okay, this is even worse, because it is now waiting for `yarn install` to complete before showing anything at all.

![image](https://github.com/iTwin/iTwinUI/assets/9084735/6255dd90-a0d5-4c4c-8f5d-969895b8f245)

The only one left worth trying is "updateContentCommand".

```
"updateContentCommand": "yarn install"
```

Reply:
tried that and its a lot faster

Reply:
I'm not seeing any change. Are you running a new codespace? Simply rebuilding is not going to be enough to test it.
![image](https://github.com/iTwin/iTwinUI/assets/9084735/337548b1-e1ff-439a-b3af-b3b08a5daf84)


One more thing we can try is changing the ["waitFor"](https://containers.dev/implementors/json_reference/#lifecycle-scripts) so that updateContentCommand runs in parallel with the UI initialization.

```
"updateContentCommand": "yarn install".
"waitFor": "onCreateCommand"
```

Reply:
Adding the updateContentCommand, and the WaitFor made it a lot faster. still have to wait for yarn install but gets to the UI sooner

Reply:
`waitFor` still isn't doing the right thing for me. The UI starts faster, but after that it waits 10-30 seconds before starting `yarn install`, which then takes 2-4 minutes to complete. I'd say lets remove `waitFor`.

Reply:
removed it

Reply:
so after comparing them all, "postCreateCommand" seems to be the best option. they all take longer than we'd like but `"onCreateCommand"` delays the UI, and `"updateCreateCommand"` has a long delay between the UI showing up and `yarn install` runnning

Reply:
Can you update your branch and change it to `pnpm install` and try again?

Reply:
`install` , `dev` and other basic worflow commands worked

Reply:
I tried it just now and it's still not a good experience. It takes too long to start `pnpm install`, as seen in the recording below


https://github.com/iTwin/iTwinUI/assets/9084735/64abb0c8-0657-4f6d-b7fc-e6646d5f78e4


Recommend changing it back.
```diff
- "postCreateCommand": "pnpm install",
- "waitFor": "onCreateCommand"
+ "onCreateCommand": "pnpm install"  
```

Reply:
made that change

Reply:
Much better! See full flow in recording below. Everything is ready in just over 1 minute.

https://github.com/iTwin/iTwinUI/assets/9084735/60cfec49-1807-4a97-8e06-6a006c4f6649